feature/sessions/src/main/java/io/github/droidkaigi/confsched2023/sessions/component/TimetableGridItem.kt

```diff
@@ -124,6 +149,48 @@ fun TimetableGridItem(
     }
 }
 
+/**
+ *
+ * Calculate the font size of the title by the height of the displayed session grid.
+ *
+ * @param textStyle session title text style.
+ * @param localDensity local density.
+ * @param gridItemHeightPx session grid item height. (unit is px.)
+ * @param speaker session speaker.
+ * @param titleLength session title length.
+ *
+ */
+private fun calculateTitleFontSize(
+    textStyle: TextStyle,
+    localDensity: Density,
+    gridItemHeightPx: Int,
+    speaker: TimetableSpeaker?,
+    titleLength: Int,
+): TextUnit {
+    val titleToScheduleSpaceHeight = with(localDensity) { 4.dp.toPx() }
+    val scheduleHeight = with(localDensity) { 16.dp.toPx() }
+    val scheduleToSpeakerSpaceHeight = with(localDensity) { 16.dp.toPx() }
+    val horizontalPadding = with(localDensity) { (12 * 2).dp.toPx() }
+
+    // The height of the title that should be displayed.
+    var displayTitleHeight =
+        gridItemHeightPx - titleToScheduleSpaceHeight - scheduleHeight - scheduleToSpeakerSpaceHeight - horizontalPadding
+    displayTitleHeight -= if (speaker != null) with(localDensity) { 32.dp.toPx() } else 0f
+
+    // Actual height of displayed title.
+    val boxWidthWithoutPadding = with(localDensity) { (192 - 12 * 2).dp.toPx() }
+    val fontSizePx = with(localDensity) { textStyle.fontSize.toPx() }
+    val textLengthInRow = (boxWidthWithoutPadding / fontSizePx).roundToInt()
+    val rows = titleLength / textLengthInRow
+    val actualTitleHeight = rows * fontSizePx
+
+    return when {
+        rows <= 1 -> textStyle.fontSize
+        displayTitleHeight > actualTitleHeight -> textStyle.fontSize
+        else -> 8.sp
```

Comment:
[Help]

We really have to calculate the appropriate size according to the session title length.

However, since the logic did not come to mind, a fixed minimum value is set.